gmcs/linglib/lexical_items.py

```diff
@@ -1014,6 +1130,30 @@ def customize_cops(mylang, ch, lexicon, hierarchies, trigger):
 
                 # TODO: Add copula types to trigger.mtr
 
+def customize_adpositions(mylang, lexicon, ch,hierarchies):
+    #If not constrained as below, prepositions will go as head daughters in head-subject rules
+    mylang.add('decl-head-subj-phrase :+ [ HEAD-DTR.SYNSEM.LOCAL.CAT.HEAD +nvj ].')
+    #mylang.add('head-spec-phrase := [ HEAD-DTR.SYNSEM.LOCAL.CAT.HEAD noun ].')
+    mylang.add(lexbase.ADP_LEX)
+    if ch.get('has-aux') == 'yes':
+        mylang.add('norm-adposition-lex := [ SYNSEM.LOCAL.CAT.HEAD.MOD < [ LOCAL.CAT.HEAD.AUX - ] > ].')
+    supertype = 'norm-adposition-lex'
+    for adp in ch.get('normadp'):
+        typename = adp.full_key + '-' + supertype
+        mylang.add(typename + ' := ' + supertype + '.')
+        for stem in adp['stem']:
+            add_stem_to_lexicon(lexicon,stem,typename)
+        features.customize_feature_values(mylang,ch,hierarchies,adp,typename,'normadp')
+        if not feature_type_use.USED_FEATURES['INIT']:
```

Comment:
You are right that this is not really serving any purpose yet. This is to start and collect statistics about which features are used by which libraries---for which this dict needs to be extended. I suppose I could clear this out from the trunk and start a develop branch for this instead.

Reply:
...Or not. Perhaps this is an example of a minimally working thing that can be merged into the trunk even if it is not serving any purpose (as opposed to it sitting in some branch for months).

Reply:
> thing that can be merged into the trunk even if it is not serving any purpose

I'd think we'd want things not serving any purpose excised from trunk.

Reply:
How is that consistent with the idea of incremental merges of unfinished libraries though?

Reply:
Are you referring to my proposal in #503? If so then I think I understand your comments above better. But to clarify, merging partial things into trunk would be better if they (1) were in their own branch and not as a rider on some other set of changes, (2) had some issue associated with them (or at least a clear and documented goal of what the changes are working toward), and (3) don't break anything. I haven't tested this, but I think it satisfies (3) but not (1) or (2). I'm concerned about experimental features being added and forgotten about. If they are grouped with some set of meaningful changes, it is hard for someone later to tease apart what is what.

That said, I like the idea of being able to quantify which features are used by which libraries, but I think it should have an issue and a branch since that effort seems orthogonal to the point of the current PR.

Reply:
I was referring to what I suggested to Emily, that it is ultimately detrimental to have any work go unmerged for a long time. And that it would be best to merge more or less anything so long as it doesn't break the tests and makes some sort of sense. I don't think lengthy discussions of whether something is fit for merge are ultimately helpful but some limited discussion certainly is! In this case, I am not sure whether it is more likely for this to get lost in the trunk or in a branch that isn't actively being developed (and to my knowledge, there is nobody in particular that is going to pick up this work soon...). That said, I will certainly open an issue for this one.

Reply:
Oh, and this PR definitely contains lots of stuff! Which is not great but it is what it is. No way of going back and turning this into a bunch of separate branches now :).

Reply:
fwiw, my personal preference is I think just a little beyond what @olzama is suggesting. I'm happy to see something merged if it doesn't break anything *and* it does something useful. I recognize "doing something useful" is a vague concept, but I think saying this particular feature's useful thing is calculating stats is not currently accurate, so I wouldn't say it passes that bar. But there's definitely a trade-off...

Reply:
> it would be best to merge more or less anything so long as it doesn't break the tests and makes some sort of sense

We should consider the [technical debt](https://en.wikipedia.org/wiki/Technical_debt). Merging partial work in has a debt to finish it. The debt is higher if someone else (not the original author) needs to finish code, especially if they are unfamiliar with the code or even its purpose. Even if it's not finished, there's a debt to decide that the code is unneeded and to remove it. In other words, if it's not immediately helping the project and it's creating work for someone else, we should probably not merge it. Note that my opinions here are more about the sentiment expressed in the quote above than this particular bit of code because I'm not entirely sure what it's for and what state it's in.

> I am not sure whether it is more likely for this to get lost in the trunk or in a branch that isn't actively being developed

I guess the point is if it is going to get lost and forgotten, it's better to be lost and forgotten in a stale branch than in trunk. Having an associated issue would help it not be forgotten.

Reply:
> I guess the point is if it is going to get lost and forgotten, it's better to be lost and forgotten in a stale branch than in trunk

But it is the opposite that is the point :). The point is to make it less likely that it is forgotten.

We have always had lots of unfinished work in the Matrix, and I don't see why it would change (that's the nature of the project; btw there are other things that I think are the nature of the project that have to do with things like conventions and protocols, but that's another topic). Given that we will always have lots of unfinished work, if we somehow make sure it always ends up in branches, we may end up with hundreds of branches a few years from now. My library alone would contribute at least a dozen of them if not more. Another library developer comes in (hopefully!) and a dozen more is added.

Honestly, I think in this appropriately commented stuff in the trunk is ultimately preferable. 

Also, I am often informed by places in the code that I stumble upon, e.g. all the unfinished stuff in information structure has been important for me to look at as I was working in wh-questions. It would not have occurred to me to search the branches. It is more about the Matrix as a living organism, if you will. Sometimes you need to get a sense of what it's like as a whole, with all the unfinished parts. Branches, I feel, are excellent for active development but not for this kind of sense.

Reply:
I can't agree with this. We're at an impasse.